torch/csrc/jit/codegen/cuda/disjoint_set.h

```diff
@@ -72,6 +74,64 @@ class DisjointSet {
     return fixedPoint(a) == fixedPoint(b);
   }
 
+  //! Queries if an element exists in this set
+  bool contains(T a) const {
+    return entry_map.count(a) > 0;
+  }
+
+  //! Just inserts an element as its own. It is considered not
+  //! equivalent with anything else.
+  bool insert(T a) {
```

Comment:
Just curious, what would be a scenario where we want to create a singleton equivalence class ,when there is no equivalence to establish. Would this be for perf optimization and debugging or there's a semantical need for this?

Reply:
Thanks. This is actually not used anymore, so I'm going to remove it.

The reason I added that was for convenience. If `a` is not in the set, `areEquivalent(a, a)` returns false, and that was not the response for a particular use case I had. 

I also thought that maybe `areEquivalent` should return always true if the two arguments are equal with `operator==`, even though the arguments are not registered. But I'm not sure what would be the best. If things aren't in the set, maybe returning false is more appropriate.

Reply:
I see, makes sense. I was originally thinking of `areEquivalent` like "this set has registered that they are equivalent", but I guess in the case where inputs are just identical in some other sense (`==`) but unseen before it's hard to say. So please feel free to use the way most convenient for you and we just want to make sure to reflect that on the API and doxygen strings.